src/Components/OATGraphViewer/Internal/OATGraphCustomEdge.tsx

```diff
@@ -50,58 +51,326 @@ const OATGraphCustomEdge: React.FC<IOATGraphCustomEdgeProps> = ({
+                false
+            );
+        }
+        return {
+            inheritancePolygon: inheritancePolygon,
+            relationshipPolygon: relationshipPolygon,
+            polygonTargetX: polygonTargetX,
+            polygonTargetY: polygonTargetY
+        };
+    };
+
+    const polygons = useMemo(() => {
+        // With this Memo function the values for Polygons Points are calculated
+        const adjustedSourceY = sourceY - sourceDefaultHeight;
+
+        const element = elements.find((x) => x.id === id);
+        let inheritancePolygon = '';
+        let relationshipPolygon = '';
+        let componentPolygon = '';
+        let polygonTargetX = 0;
+        let polygonTargetY = 0;
+        let polygonSourceX = 0;
+        let polygonSourceY = 0;
+        if (element) {
+            // If a valid element we get size based in positioning
+            const sourceNode = elements.find((x) => x.id === element.source);
+            const sourceNodeSizeX = (sourceX - sourceNode.position.x) * 2;
+            const sourceNodeSizeY =
+                (adjustedSourceY - sourceNode.position.y) * 2;
+            const targetNode = elements.find((x) => x.id === element.target);
+            const targetNodeSizeX = (targetX - targetNode.position.x) * 2;
+            const targetNodeSizeY = (targetY - targetNode.position.y) * 2;
+            // Getting vectors to adjust angle from source to target
+            let heightVector = targetY > adjustedSourceY ? 1 : -1;
+            let baseVector = targetX > sourceX ? 1 : -1;
+            // Using source and target points to triangulate and get angles
+            const triangleHeight = (targetY - adjustedSourceY) * heightVector;
+            const triangleBase = (targetX - sourceX) * baseVector;
+            const triangleHypotenuse = Math.sqrt(
+                triangleHeight * triangleHeight + triangleBase * triangleBase
+            );
+            // Using source size to triangulate connection with source edge
+            const sourceHeight = sourceNodeSizeY / 2;
+            const sourceBase = sourceNodeSizeX / 2;
+            const sourceHypotenuse = Math.sqrt(
+                sourceHeight * sourceHeight + sourceBase * sourceBase
+            );
+            const sourceBetaAngle = Math.asin(sourceBase / sourceHypotenuse);
+            const alphaAngle = Math.asin(triangleHeight / triangleHypotenuse);
+            const betaAngle = rightAngleValue - alphaAngle;
+            const sourceComponents = getSourceComponents(
+                betaAngle,
+                sourceBase,
+                sourceBetaAngle,
+                sourceHeight,
+                alphaAngle,
+                adjustedSourceY,
+                baseVector,
+                heightVector
+            );
+            componentPolygon = sourceComponents.componentPolygon;
+            polygonSourceX = sourceComponents.polygonSourceX;
+            polygonSourceY = sourceComponents.polygonSourceY;
```

Comment:
We could do something like: `polygons = { polygons, ...sourceComponents };` (i.e. rather than setting everything using `let` at the start, it could just be `let polygons = {};`.